### PR TITLE
♻️  [bento][amp-stream-gallery] Split component files for NPM Packaging

### PR DESCRIPTION
**Pull request by krdwan**
Tracker Issue: https://github.com/ampproject/amphtml/issues/34137

Split amp-stream-gallery for NPM Packaging.

### CODE REVIEW

**Review by rcebulko — COMMENTED** [review]
Is there an entry for this in `bundles.config.js`?

File: extensions/amp-stream-gallery/1.0/test/test-stream-gallery.js [hunk 1]
```diff
@@ -15,7 +15,7 @@
  */
 
 import * as Preact from '../../../../src/preact';
-import {StreamGallery} from '../stream-gallery';
+import {StreamGallery} from '../component';
```
Comment (caroqliu) [inline extensions/amp-stream-gallery/1.0/test/test-stream-gallery.js:18]: Please rename this file as `test-component.js`

File: extensions/amp-stream-gallery/1.0/base-element.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS as CAROUSEL_CSS} from '../../amp-base-carousel/1.0/base-carousel.jss';
+import {CSS as GALLERY_CSS} from './stream-gallery.jss';
```
Comment (caroqliu) [inline extensions/amp-stream-gallery/1.0/base-element.js:18]:
```suggestion
import {CSS as GALLERY_CSS} from './component.jss';
```

**Review by caroqliu — APPROVED** [review]

**Review by jridgewell — APPROVED** [review]

**Comment by krdwan** [comment]
> Is there an entry for this in `bundles.config.js`?

Thanks for the recommendation, I've added it in the latest revision.

File: extensions/amp-stream-gallery/1.0/test/test-stream-gallery.js (hunk 1 above)
Comment (krdwan) [reply extensions/amp-stream-gallery/1.0/test/test-stream-gallery.js:18]: Thanks, updated!

File: extensions/amp-stream-gallery/1.0/base-element.js (hunk 2 above)
Comment (krdwan) [reply extensions/amp-stream-gallery/1.0/base-element.js]: Thanks, updated in the latest revision!

**Comment by krdwan** [comment]
Added the `package.json` file which is needed for manual publishing.

**Comment by krdwan** [comment]
Updated `dep-check-config.js` which is needed for dependencies across extensions.

**Review by jridgewell — APPROVED** [review]

**Review by rcebulko — APPROVED** [review]
LGTM!

File: build-system/compile/bundles.config.extensions.json [hunk 3]
```diff
@@ -646,9 +646,25 @@
...
+    "version": "1.0",
+    "latestVersion": "0.1",
+    "options": {
+      "hasCss": true,
+      "npm": {
+        "component.js": {
+          "preact": "component-preact.js",
+          "react": "component-react.js"
+        }
+      }
```
Comment (rcebulko) [inline build-system/compile/bundles.config.extensions.json:666]:
```suggestion
      "npm": true
```

Comment (krdwan) [reply build-system/compile/bundles.config.extensions.json]: Awesome, thanks!  I saw that you just merged your change this morning!

**Pull request merged**